### ISSUE TITLE
Panic on v0.22.5 cause by bad go-amino unmarshalling

### ISSUE THREAD

**Issue by silasdavis** [tendermint/tendermint#2063]
After upgrading Tendermint from 0.22.4 to 0.22.5 I got a panic in fairly simple test that boots a node and shuts it down: https://github.com/hyperledger/burrow/blob/develop/integration/core/kernel_test.go#L29-L36

This test is similar in spirit to your `TestNodeStartStop` so I compared the two. I was able to pin down the issue to `state.LastValidators` coming back nil (rather than just empty when I call it as opposed to where your test calls it. I am `TestNodeStartStop` from my own vendor so all versions including amino are the same. I changed the `Bytes()` method code in state/state.go to:

```go
// Bytes serializes the State using go-amino.
func (state State) Bytes() []byte {
	bs := cdc.MustMarshalBinaryBare(state)
	stateOut := new(State)
	cdc.MustUnmarshalBinaryBare(bs,stateOut)
	fmt.Println("State in:")
	fmt.Println(state)
	fmt.Println("\nState out:")
	fmt.Println(stateOut)
	return bs
}
```
In `TestNodeStopStart` this gives:
```
State in:
{tendermint_test 0 0 :0:000000000000 2018-07-25 11:36:20.377603398 +0100 BST m=+0.006137928 ValidatorSet{
  Proposer: Validator{A3258DCBF45DCA0DF052981870F2D1441A36D145 PubKeyEd25519{013FFE69A2F5781D38EFB32E77D24C9BC4A1F0122F39894F182F4A90852615A1} VP:10 A:0}
  Validators:
    Validator{A3258DCBF45DCA0DF052981870F2D1441A36D145 PubKeyEd25519{013FFE69A2F5781D38EFB32E77D24C9BC4A1F0122F39894F182F4A90852615A1} VP:10 A:0}
} ValidatorSet{
  Proposer: nil-Validator
  Validators:
    
} 1 {{22020096 10000 -1} {10240 -1} {65536} {100000}} 1 [] []}
State out:
&{tendermint_test 0 0 :0:000000000000 2018-07-25 10:36:20.377603398 +0000 UTC ValidatorSet{
  Proposer: Validator{A3258DCBF45DCA0DF052981870F2D1441A36D145 PubKeyEd25519{013FFE69A2F5781D38EFB32E77D24C9BC4A1F0122F39894F182F4A90852615A1} VP:10 A:0}
  Validators:
    Validator{A3258DCBF45DCA0DF052981870F2D1441A36D145 PubKeyEd25519{013FFE69A2F5781D38EFB32E77D24C9BC4A1F0122F39894F182F4A90852615A1} VP:10 A:0}
} ValidatorSet{
  Proposer: nil-Validator
  Validators:
    
} 1 {{22020096 10000 -1} {10240 -1} {65536} {100000}} 1 [] []}
```
Note that the `LastValidators` `ValidatorSet` comes back as an empty but non-nil object.

When I run it from my kernel_test.go I get:

```
State in:
{TestChain-219ABF 0 0 :0:000000000000 2017-09-23 13:07:17 +0000 UTC ValidatorSet{
  Proposer: Validator{BA5AD671EDA23911C6B5D1B0F8A98E9B7D62C4BB PubKeyEd25519{71A85B72A2BC92D107CB0A5999194478DF1FD44E19720CCECABED3051F9859A4} VP:4139441442680080991 A:0}
  Validators:
    Validator{BA5AD671EDA23911C6B5D1B0F8A98E9B7D62C4BB PubKeyEd25519{71A85B72A2BC92D107CB0A5999194478DF1FD44E19720CCECABED3051F9859A4} VP:4139441442680080991 A:0}
} ValidatorSet{
  Proposer: nil-Validator
  Validators:
    
} 1 {{22020096 10000 -1} {10240 -1} {65536} {100000}} 1 [] [33 154 191 172 42 159 27 217 92 155 198 231 217 142 238 83 21 59 169 31 48 108 112 134 167 34 64 121 226 218 182 48]}

State out:
&{TestChain-219ABF 0 0 :0:000000000000 2017-09-23 13:07:17 +0000 UTC ValidatorSet{
  Proposer: Validator{BA5AD671EDA23911C6B5D1B0F8A98E9B7D62C4BB PubKeyEd25519{71A85B72A2BC92D107CB0A5999194478DF1FD44E19720CCECABED3051F9859A4} VP:4139441442680080991 A:0}
  Validators:
    Validator{BA5AD671EDA23911C6B5D1B0F8A98E9B7D62C4BB PubKeyEd25519{71A85B72A2BC92D107CB0A5999194478DF1FD44E19720CCECABED3051F9859A4} VP:4139441442680080991 A:0}
} nil-ValidatorSet 1 {{22020096 10000 -1} {10240 -1} {65536} {100000}} 1 [] [33 154 191 172 42 159 27 217 92 155 198 231 217 142 238 83 21 59 169 31 48 108 112 134 167 34 64 121 226 218 182 48]}
```
See that `LastValidators` comes back as nil not just empty: `nil-ValidatorSet`

This later causes a panic on `state.Copy()`.

The only think I can think of that could cause this is some global state in go-amino - I am currently also using go-amino in Burrow which involves calling `aminoCrypto.Register` and some other registrations. Could this somehow cause it. Either that or some difference in the `GenesisDoc`. I notice my time is BST... (??)

**Comment by xla** [comment]
@silasdavis We release `v0.22.6` as a hotfix, can you try and see if it addresses the issue?

**Comment by silasdavis** [comment]
I did earlier today and it did not.

**Comment by ValarDragon** [comment]
Closing this for now. Please comment / re-open if this problem persists on `v0.22.8` for you.

**Comment by silasdavis** [comment]
fixed for me in 0.22.8
